homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,297 @@
+
+
+class UpdateManager:
+    """Update manager for the update integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the update manager."""
+        self._hass = hass
+        self._store = storage.Store(
+            hass=hass,
+            version=STORAGE_VERSION,
+            key=DOMAIN,
+        )
+        self._skip: set[str] = set()
+        self._updating: set[str] = set()
+        self._registrations: dict[str, UpdateRegistration] = {}
+        self._updates: dict[str, list[UpdateDescription]] = {}
+
+    @property
+    def filtered_updates(self) -> list[dict]:
+        """Return a list of updates that are not skipped."""
+        return [
+            {
+                "domain": domain,
+                "identifier": description.identifier,
+                "name": description.name,
+                "current_version": description.current_version,
+                "available_version": description.available_version,
+                "changelog_url": description.changelog_url,
+                "icon_url": description.icon_url,
+                "provides_backup": description.provides_backup,
+            }
+            for domain, domain_data in self._updates.items()
+            if domain_data is not None
+            for description in domain_data
+            if f"{domain}_{description.identifier}_{description.available_version}"
+            not in self._skip
+        ]
+
+    async def load(self) -> None:
+        """Load the stored data."""
+        self._skip = set(await self._store.async_load() or [])
+
+    async def gather_updates(self) -> list[dict]:
+        """Gather updates."""
+        for domain, domain_data in zip(
+            self._registrations,
+            await asyncio.gather(
+                *(
+                    self._get_integration_info(registration)
+                    for registration in self._registrations.values()
+                )
+            ),
+        ):
+            self._updates[domain] = domain_data
+        return self.filtered_updates
+
+    @callback
+    def _data_to_save(self) -> list:
+        """Schedule storing the data."""
+        return list(self._skip)
+
+    def _get_update_description(
+        self, domain: str, identifier: str
+    ) -> UpdateDescription | None:
+        """Get an update."""
+        return next(
+            (
+                update
+                for update in self._updates.get(domain, [])
+                if update.identifier == identifier
+            ),
+            None,
+        )
+
+    def register(self, registration: UpdateRegistration) -> None:
+        """Register an update handler."""
+        self._registrations[registration.domain] = registration
+
+    async def perform_update(
+        self,
+        domain: str,
+        identifier: str,
+        **kwargs: Any,
+    ) -> bool | None:
+        """Perform an update."""
+        update_description = self._get_update_description(domain, identifier)
```

Comment:
We don't really have to store this, do we ? We could just call a method on the update platform to update?